packages/uma-auth-client/src/hooks/useOAuth.tsx

```diff
@@ -0,0 +1,100 @@
+import {auth, nwc, webln} from '@getalby/sdk';
+import { useEffect, useState } from "react";
+import { getUmaDomain } from "src/utils/getUmaDomain";
+import { useUser } from "./useUser";
+import { create } from 'zustand';
+import { persist } from "zustand/middleware";
+import { getNwcUmaDomain } from 'src/utils/getNwcUmaDomain';
+
+type UmaAuthToken = {
+  access_token: string,
+  refresh_token: string,
+  expires_at: number,
+  nwc_connection_uri: string;
+  commands: string[];
+  budget: string;
+  nwc_expires_at: string;
+};
+
+interface TokenState {
+  accessToken: string;
+  refreshToken: string;
+  expiresAt: number;
+}
+
+interface OAuthState {
+  codeVerifier?: string;
+  token?: TokenState;
+  identityNpub?: string;
+  identityRelayUrl?: string;
+  redirectUri?: string;
+  nwcConnectionUri?: string;
+  setToken: (token: TokenState) => void;
+  performOAuth: (uma: string) => Promise<void>;
+}
+
+export const useOAuth = create<OAuthState>()(
+  persist(
+    (set, get) => ({
+      setToken: (token) => set({
```

Comment:
It's probably pretty important for client apps to be able to delete this state too. Should we add a clearToken or make token nullable?

Reply:
Hm would need to expose a logout function somehow to the client app in that case... updated setToken to take an undefined token for now